app/controllers/authenticated_controller.rb

```diff
@@ -0,0 +1,11 @@
+class AuthenticatedController < ApplicationController
+    before_action :check_user, :set_user
+
+    def check_user
+        redirect_to root_path if (session[:current_user].nil? || User.find_by(id: session[:current_user]).nil?)
```

Comment:
nit: is the first part of the if statement necessary? Would find_by just return nil if id is nil?